aiida_mlip/data/model.py

```diff
@@ -0,0 +1,159 @@
+"""Define Model Data type in AiiDA"""
+
+import hashlib
+from pathlib import Path
+from urllib import request
+from urllib.parse import urlparse
+
+from aiida.orm import SinglefileData
+
+
+def _calculate_hash(file: str | Path) -> str:
+    """Function to calculate file hashes
+    param: file: path to the file you want to calculate the hash for (str)
+    """
+    # Calculate hash
+    buf_size = 65536  # reading 64kB (arbitrary) at a time
+    sha256 = hashlib.sha256()
+    with open(file, "rb") as f:
+        # calculating sha in chunks rather than 1 large pass
+        while data := f.read(buf_size):
+            sha256.update(data)
+    file_hash = sha256.hexdigest()
+    return file_hash
+
+
+def _check_existing_file(
```

Comment:
Out of interest, is there a benefit to having this function and `_calculate_hash` defined outside of the class?